src/NuGet.Core/NuGet.Protocol/Plugins/Messages/GetAuthenticationCredentialsRequest.cs

```diff
@@ -37,11 +43,12 @@ public sealed class GetAuthenticationCredentialsRequest
         /// <param name="isNonInteractive"></param>
         /// <exception cref="ArgumentNullException"> if <paramref name="uri"/> is null</exception>
         [JsonConstructor]
-        public GetAuthenticationCredentialsRequest(Uri uri, bool isRetry, bool isNonInteractive)
+        public GetAuthenticationCredentialsRequest(Uri uri, bool isRetry, bool isNonInteractive, bool canShowDialog)
```

Comment:
could there be a request where `IsNonInteractive` is `false` but `canShowDialog` is `true`? Like this request is non interactive but can still show dialog for user credential inputs?

Reply:
Yes, but it doesn't make sense.

canShowDialog talks about the authentication method. 
Whether it's device flow vs basic auth/2-factor with a pop-up. 

Interactive is whether a plugin can block the build to authenticate or it should use the cached credentials only. 

I will update the class members with more details and better descriptions.

Reply:
that's what I'd imagine so should we make sure if `IsNonInteractive` is `false` then `canShowDialog` is also implicitly made `false`?

Reply:
Ehhh..I'm not a fan of changing the values passed in a constructor like that :) 
This is a model used for transport only, it shouldn't contain business logic like that. 

Unless you have other ideas/disagree, I'd rather leave it like that.

Reply:
I see this as a loophole in current impl, where even though interactive flag is off, then also it could show dialog which could hang your build or operation running in CI. I leave it up to however you want to resolve it.

Reply:
It would be a loophole in any cross process plugin implementation, because we have no way of enforcing that the plugin cooperates the way it should (they could just ignore that flag). 
It was the same in previous plugin implementation.

Interactive is what blocks the build. Whether the build is blocked with a dialog or with a message asking for user action is not particularly relevant.

Reply:
No I dont think I follow through... I think we have a way to enforce that by making sure that when interactive is false, then showDialog is also false, which shouldn't be a big deal to achieve.